### PR TITLE
add: workflow to run smithy-rs sync tool

### PR DESCRIPTION
**Pull request by Velfi**
*Issue #, if available:*
smithy-rs#928

*Description of changes:*
This workflow runs [this tool](https://github.com/awslabs/aws-sdk-rust/pull/362) every hour and commits the changes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/smithy-rs-sync.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        default: true
+    - name: build sync tool
+      run: cargo build --release
+      working-directory: smithy-rs/tools/smithy-rs-sync
+    - name: run sync tool
+      run: |
+        smithy-rs/target/release/smithy-rs-sync \
+          --smithy-rs smithy-rs \
+          --aws-sdk aws-sdk-rust
+    - name: push commits
```
Comment (rcoh) [inline .github/workflows/smithy-rs-sync.yml:35]: any reason to not do `cargo run --release`? TBH, not a huge reason to run in release mode, not like there's any performance benefit

**Review by rcoh — APPROVED** [review]

File: .github/workflows/smithy-rs-sync.yml (hunk 1 above)
Comment (Velfi) [reply .github/workflows/smithy-rs-sync.yml:35]: I'm fine switching it over, I think I'll just need to add `../` to the paths.

File: .handwritten_files [hunk 2]
```diff
@@ -18,12 +18,10 @@ _config.yml
 .gitignore
 .handwritten_files
 .smithyrs-githash
```
Comment (rcoh) [inline .handwritten_files:20]: this is autogenerated too right?

Comment (Velfi) [reply .handwritten_files:20]: good catch! I wasn't committing this for anything but the last commit and I really should be doing it for every commit, let me fix that.

**Pull request merged**